### ISSUE TITLE
regex-lite with a &[u8] haystack

### ISSUE THREAD

**Issue by SimonSapin** [rust-lang/regex#1196]
#### Describe your feature request

[regex::bytes::Regex](https://docs.rs/regex/latest/regex/bytes/struct.Regex.html) can search `&[u8]` haystacks that are not necessarily well-formed UTF-8. This is great for text-based file formats that predate Unicode and hand-wave encodings as a platform detail.

Could the `regex-lite` crate have a similar feature? As far as I can tell it doesn’t in 0.1.5

**Comment by BurntSushi** [comment]
Yeah that is definitely the intent. The main internal APIs are [specifically defined on `&[u8]`](https://github.com/rust-lang/regex/blob/ddeb85eaa3bdf79d6306cc92a9d8bd89d839b5cd/regex-lite/src/pikevm.rs#L72). I just didn't do this initially because I wasn't 100% certain folks would want it.

The main hitch here is that I think it needs to be a disabled by default opt-in feature. The reason is that it will add a fair bit of code (basically a copy of `string.rs`, but for bytes), and the primary purpose of `regex-lite` is to keep binary size small and compilation times short.

I'm not sure when I'll have a chance to work on this, but I could review a PR adding this.

**Comment by SimonSapin** [comment]
Thanks for the feedback. For what it’s worth, for now I went with `regex::bytes` and disabled Unicode-related cargo features.
